src/ObjCRuntime/Runtime.CoreCLR.cs

```diff
@@ -77,13 +80,45 @@ unsafe struct MonoMethodSignature {
 		static bool? track_monoobject_with_stacktraces;
 #endif
 
-		// Comment out the attribute to get all printfs
-		[System.Diagnostics.Conditional ("UNDEFINED")]
+		// Define VERBOSE_LOG at the top of this file to get all printfs
+		[System.Diagnostics.Conditional ("VERBOSE_LOG")]
 		static void log_coreclr (string message)
 		{
 			NSLog (message);
 		}
 
+		// Define VERBOSE_LOG at the top of this file to get all printfs
+		[System.Diagnostics.Conditional ("VERBOSE_LOG")]
+		static void log_coreclr_render (string message, params object[] argumentsToRender)
+		{
+			var args = new string[argumentsToRender.Length];
+			for (var i = 0; i < args.Length; i++) {
+				string arg;
+				var obj = argumentsToRender [i];
+				if (obj is null) {
+					arg = "<null>";
+				} else if (obj is IntPtr ptr) {
+					arg = $"0x{ptr.ToString ("x")} (IntPtr)";
+				} else if (obj.GetType ().IsValueType) {
+					arg = $"{obj.ToString ()} ({obj.GetType ()})";
+				} else if (obj is INativeObject inativeobj) {
+					// Don't call ToString on an INativeObject, we may end up with infinite recursion.
+					arg = $"{inativeobj.Handle.ToString ()} ({obj.GetType ()})";
+				} else {
+					var toString = obj.ToString();
+					var eol = toString.IndexOf ('\n');
+					if (eol != -1)
+						toString = toString.Substring (0, eol - 1) + " [...]";
```

Comment:
If `ToString` returns "\nSomeText", this will throw `ArgumentOutOfRangeException`.
Maybe instead you want to instead do:
```
if (eol != -1)
    toString = toString.Replace("\n", "\\n");
```
And let the 256 trim take care of everything else.

Reply:
I reworked this a little bit to fix the issue you caught and also only call Substring once